src/vs/workbench/services/positronConsole/browser/positronConsoleService.ts

```diff
@@ -2124,6 +2160,7 @@ class PositronConsoleInstance extends Disposable implements IPositronConsoleInst
 						// are currently in the Offline state, the message that
 						// brings us back online may not be one of our own messages.
 						if (languageRuntimeMessageState.parent_id.startsWith('fragment-') ||
+							this._externalExecutionIds.has(languageRuntimeMessageState.parent_id) ||
```

Comment:
Do we still need the `fragment-` check if we explicitly track external execution IDs?

Reply:
Yes, because internally generated IDs (`fragment-...`) don't get tracked this way. Of course we could also just stop relying on naming patterns and track everything in a set, but I didn't want to refactor that much!